randutil/rand.go

```diff
@@ -23,21 +23,41 @@
 package randutil
 
 import (
-	cryptorand "crypto/rand"
-	"fmt"
-	"math"
-	"math/big"
+	"crypto/sha256"
+	"encoding/binary"
 	"math/rand"
+	"net"
+	"os"
+	"sync"
 	"time"
 )
 
 func init() {
 	// golang does not init Seed() itself
-	bigSeed, err := cryptorand.Int(cryptorand.Reader, big.NewInt(math.MaxInt64))
-	if err != nil {
-		panic(fmt.Sprintf("cannot obtain random seed: %v", err))
-	}
-	rand.Seed(bigSeed.Int64())
+	rand.Seed(time.Now().UnixNano() + int64(os.Getpid()))
+}
+
+var moreMixedSeedOnce sync.Once
+
+func moreMixedSeed() {
+	moreMixedSeedOnce.Do(func() {
+		h := sha256.New224()
+		// do this instead of asking for time and pid again
+		var b [8]byte
+		rand.Read(b[:])
+		h.Write(b[:])
+		if hostname, err := os.Hostname(); err == nil {
```

Comment:
Should we mix in time.Now().UnixNano() and os.Getpid() into the hash too for good measure?

Reply:
we used them when we created the original seed,  I'm instead of doing that again, mixing in 8 bytes out of the current sequence , see the lines just before this one, that's what the comment tries to say